src/controllers/usersController.ts

```diff
@@ -55,12 +56,20 @@ const handleUserRequestById = async (id: string, selfInfo: boolean, request: any
           builder.where('id', id);
         });
 
-    const [contacts, user] = await Promise.all([contactsPromise, userPromise]);
+    const isFriendPromise = selfInfo ? null :
+      await FriendEntries()
+          .select('id')
+          .first()
+          .where('userId', request.user!.id!)
+          .andWhere('friendId', id);
+
+    const [contacts, user, isFriend] = await Promise.all([contactsPromise, userPromise, isFriendPromise].filter(t => t));
```

Comment:
Проблема с Promise.all если один из промисов завалится при исполнении, то оставшиеся могут не отработать. 
Вот что пишет на эту тему MDN:
`
Promise.all() will reject immediately upon any of the input promises rejecting. In comparison, the promise returned by Promise.allSettled() will wait for all input promises to complete, regardless of whether or not one rejects. Consequently, it will always return the final result of every promise and function from the input iterable.
`
Так что лучше заменить на **Promise.allSettled()**

Reply:
почему это проблема? если часть промисов завалится, то нам нет смысла возвращать полуразбитый результат. нам нужно иметь все заполненые свойства.

Reply:
Для того, чтобы точно трекать, сколько у нас лямбда каждый запрос обрабатывает - неплохо не оставлять за собой запросы к базе данных, которые еще в процессе. Иначе на первый reject мы вроде бы как доделаем все и вернем результат, но на самом деле оставим за собой еще какую-то работу, которая продолжится.
Если бы у нас был бы магический примитив, который бы убивал бы другие промизы, когда хотя бы один умер, то да - Promise.all в данном случае хороший, но у нас его к сожалению нет и cancelable promises это все еще не решенный в спеке вопрос. 
`Promise.allSettled` если что на aws lambda работает.

Reply:
Я не смог найти где было сказано, что они будут продолжать charging за HTTP-request, который уже обработан функцией и вернули response. Но спорить не буду, проще поменять.